src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int currentSize;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementData[currentSize++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, currentSize);
+        growIfArrayFull();
+        if (index == currentSize) {
+            elementData[currentSize++] = value;
+            return;
+        }
+        int tailLength = currentSize - index;
+        System.arraycopy(elementData, index, elementData, index + 1, tailLength);
+        elementData[index] = value;
+        currentSize++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        int minLength = Math.max(list.size() + currentSize, DEFAULT_CAPACITY);
+        grow(minLength);
+        for (int i = 0; i < list.size(); i++) {
+            elementData[currentSize++] = list.get(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, currentSize - 1);
+        return (T) elementData[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index, currentSize - 1);
+        elementData[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, currentSize - 1);
+        T elemToRemove = (T) elementData[index];
+        if (index == currentSize) {
+            elementData[index] = null;
+            return elemToRemove;
+        }
+        int tailLength = currentSize - index - 1;
+        System.arraycopy(elementData, index + 1, elementData, index, tailLength);
+        currentSize--;
+        return elemToRemove;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < currentSize; i++) {
+            if (Objects.equals(element, elementData[i])) {
```

Comment:
plz check [this](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-use-class-objects)